ARROW-13730: Adding a column to an existing Table (#81)
apache/arrow-cookbook: python/source/data.rst

  cast data from one type to another (if `promote=True`).  In such cases the data 
  will need to be copied and an extra cost will occur.

Adding a column to an existing Table
====================================

If you have a table it is possible to extend its columns using
:meth:`pyarrow.Table.append_column`

> Suppose we have a table with oscar nominations for each actress

Reviewer: ```suggestion
Suppose we have a table with Oscars won by each actress
```

Reviewer 2: Are you sure that's the case? I doubt a single person won 21 oscars. 21 seems to be the number of nominations for Meryl Streep.

Reviewer: Oh, I misread and thought @amol- was using the same dataset as the one from the R cookbook, but you're right - this is a different one.

.. testcode::

  import pyarrow as pa

  oscar_nominations = pa.table([
    ["Meryl Streep", "Katharine Hepburn"],
    [21, 12]

>   ], names=["actor", "nominations"])

Reviewer 2: Nit: do you want to say "actress"?

  print(oscar_nominations)

.. testoutput::

  pyarrow.Table
  actor: string

>   nominations: int64

Reviewer 2: Do you want to update this now you have added a summary to the Table repr?
